src/test/test-integration.ts

```diff
@@ -0,0 +1,95 @@
+
+/// <reference path="../../typings/knockout/knockout.d.ts" />
+/// <reference path="../../typings/jasmine/jasmine.d.ts" />
+/// <reference path="../../typings/socket.io-client/socket.io-client.d.ts" />
+
+import { App } from "../common/client-app";
+import * as viewModel from "../common/view-model";
+import * as model from "../common/model";
+import * as ko from "knockout";
+import { BoardVM } from "../koclient/ko-view-model";
+import {BoardService} from "../server/BoardService";
+
+describe("Integration test between client and service", function(){
+// TODO: maybe separate this two classes in a module
+
+  class ServerMock{
+  private _boardService: BoardService;
+  private _clients: Array<App>;
+  constructor(boardService:BoardService){
+    this._boardService = boardService;
+    this._clients = new Array<App>();
+    var server = this;
+    this._boardService.sendToClient = (changes) => {
+      for(var client in server._clients){
+        server._clients[client].onMessage({patch: changes});
+      }
+    };
+  }
+
+  public addClient = (client: App) => {
+    var server = this;
+    client.sendToServer = (changes) => {
+      server._boardService.onClientMessage(changes);
```

Comment:
Since you are using lambda function syntax, I think that `server` variable is not necessary. If it is compiled to ES5, TypeScript generates the variable automatically.